args.go

```diff
@@ -33,15 +34,15 @@ func legacyArgs(cmd *Command, args []string) error {
 
 	// root command with subcommands, do subcommand checking.
 	if !cmd.HasParent() && len(args) > 0 {
-		return fmt.Errorf("unknown command %q for %q%s", args[0], cmd.CommandPath(), cmd.findSuggestions(args[0]))
+		return fmt.Errorf(gotext.Get("LegacyArgsValidationError"), args[0], cmd.CommandPath(), cmd.findSuggestions(args[0]))
```

Comment:
How about leaving the original english string as the index for all the `gotext.Get()` calls?
It would make reading the code much easier and it would make working with the `*.po` files easier as the original text would be right there as the index.

If the `%` formatting gives a problem, maybe we can replace it with `%%` in the index?  It is not ideal, but it would be manageable.  So, for this line here we would instead use (unless there is a better way?)
```
return fmt.Errorf(gotext.Get("unknown command %%q for %%q%%s"), args[0], cmd.CommandPath(), cmd.findSuggestions(args[0]))

```

Reply:
I’ve looked I’d gotext in more detail, and the `%` form will work just fine as long as the parameters are in the `gotext.Get()` call instead of outside.

Reply:
This is the crux of the matter.

I've done a whole bunch of implementations of translations in the past, and every time I try to use plain english instead of keys I end up either regretting it or refactoring heavily to keys.

Here's some food for thought when using raw english as keys:

- cosmetically changing the english string (typo, whitespace, etc.) invalidates all the existing translations (possibly hundreds of files need to be updated, for nothing)
- compels usage of `context` shenanigans for strings that are the same in english in different contexts but different in other languages
- too many gettext parsers out there choke on nasty translation ids like `the "quoted q'tara" <tag>`
- sometimes it creates a bias towards english in the structure of strings, that need to be fixed by translators, straight in the code, once again invalidating all existing translations

-----

I'm very aware that using keys makes the code harder to read and understand.  This is mitigated a little by a careful choice of the wording of the key.

I had to pick one way or the other ; it was not an easy choice, but it's one I made many times and I decided to go with hindsight from past experiences.

I'm not adamant on this, quite the contrary.
I listed some of the key points of my decision above (probably forgot some) ; I'll let y'all be the final judges.   Good luck !